src/Model/CommonTypes.hs

```diff
@@ -7,20 +7,36 @@ import Graphics.Gloss.Interface.IO.Game (Key, KeyState(..))
 class UnwrapablePair a where
   unwrap :: a -> (Float, Float)
 
+class LSE a where
```

Comment:
Define LSE

Reply:
done

Reply:
It is too complicated for the task IMHO

Reply:
And `LSE` is too arbitrary and will definitely lead to errors and/or confusion in the code.